File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operations = {"+": lambda a,b: a+b, "-": lambda a,b: a-b, "*": lambda a,b: a * b, "/": lambda a,b: int(a/b)}
        
        for char in tokens:
            if char in "+-*/":
                b = stack.pop()
                a = stack.pop()
                stack.append(operations[char](a,b))
            else:
                stack.append(int(char))
        
        return stack[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.rb
# @param {String[]} tokens
# @return {Integer}
def eval_rpn(tokens)
    operations = {
        "+" => lambda { |a, b| a + b },
        "-" => lambda { |a, b| a - b },
        "*" => lambda { |a, b| a * b },
        "/" => lambda { |a, b| (a.to_f/b).to_i},
    }
    stack = []
    tokens.each do |str|
        if operations[str]
            b = stack.pop
            a = stack.pop

            stack.push(operations[str].call(a,b))
        else
            stack.push(str.to_i)
        end
    end

    return stack.pop
end